Modules/Filtering/ImageFeature/include/itkLaplacianSharpeningImageFilter.hxx

```diff
@@ -183,51 +98,41 @@ LaplacianSharpeningImageFilter<TInputImage, TOutputImage>::GenerateData()
     ++it;
     ++inIt;
   }
-  RealType inputMean = inputSum / static_cast<RealType>(this->GetOutput()->GetRequestedRegion().GetNumberOfPixels());
-  RealType enhancedMean =
-    enhancedSum / static_cast<RealType>(this->GetOutput()->GetRequestedRegion().GetNumberOfPixels());
+  RealType numberOfVoxels = static_cast<RealType>(this->GetInput()->GetRequestedRegion().GetNumberOfPixels());
+  RealType inputMean = inputSum / numberOfVoxels;
+  RealType enhancedMean = enhancedSum / numberOfVoxels;
+
+  auto inputMinimum = static_cast<RealType>(inputCalculator->GetMinimum());
+  auto inputMaximum = static_cast<RealType>(inputCalculator->GetMaximum());
 
-  // update progress
-  this->UpdateProgress(0.9);
+  // Allocate and write the output
 
-  // copy and cast the output
   typename TOutputImage::Pointer output = this->GetOutput();
-  output->SetBufferedRegion(output->GetRequestedRegion());
+  output->CopyInformation(this->GetInput());
+  output->SetRegions(output->GetRequestedRegion());
   output->Allocate();
 
-  RealType inputMinimum = inputCalculator->GetMinimum();
-  RealType inputMaximum = inputCalculator->GetMaximum();
-  auto     castInputMinimum = static_cast<OutputPixelType>(inputMinimum);
-  auto     castInputMaximum = static_cast<OutputPixelType>(inputMaximum);
-
   ImageRegionIterator<OutputImageType> outIt(output, output->GetRequestedRegion());
   it.GoToBegin();
   while (!outIt.IsAtEnd())
   {
-    value = it.Get();
-
     // adjust value to make the mean intensities before and after match
```

Comment:
The UnaryGeneratorImageFilter can except lambda functions which should make converting the code to run multi-threaded in a filter relatively easy.